### PR TITLE
Drop support for Python version 2.7.

### PR DESCRIPTION
**Pull request by csadorf**
Remove code paths and testing related to support for Python 2.7.

### CODE REVIEW

**Comment by csadorf** [comment]
There might be other parts within the code that could benefit from Python 3 features, I just did a relatively shallow search for the term "2.7" to find the obvious spots.

**Comment by vyasr** [comment]
Can we remove `six` itself?

**Comment by bdice** [comment]
> 
> 
> Can we remove `six` itself?

See #158, I just did. 😄

File: changelog.txt [hunk 1]
```diff
@@ -5,6 +5,12 @@ Changes
 The **signac-flow** package follows `semantic versioning <https://semver.org/>`_.
 The numbers in brackets denote the related GitHub issue and/or pull request.
 
+next
+----
+
+- Drop support for Python version 2.7.
```
Comment (bdice) [inline changelog.txt]:
```suggestion
Removed
+++++++
- Drop support for Python version 2.7.
```

**Pull request merged**